cesium/internal/index/domain.go

```diff
@@ -133,11 +144,27 @@ func (i *Domain) Stamp(
 		return
 	}
 
+	// endOffset is the upper-bound distance of the desired sample from the start of the
+	// domain.
 	endOffset := startApprox.Upper + offset
+
+	// If the upper and lower bounds are exact of the startOffset are exact, then if the
+	// lower is out of the file, the stamp is discontinuous.
+	// If they are not exact, and the lower bound is the last sample, then the upper
+	// bound must be discontinuous as well.
+	if continuous {
+		if (startApprox.Exact() && startApprox.Lower+offset >= effectiveDomainLen/8) ||
+			(!startApprox.Exact() && startApprox.Lower+offset >= effectiveDomainLen/8-1) {
+			err = ErrDiscontinuous
+			return
+		}
+	}
+
 	gap := iter.Len() / 8
 	if endOffset >= iter.Len()/8 {
 		for {
 			if !iter.Next() {
+				// exhausted
 				if continuous {
 					err = ErrDiscontinuous
```

Comment:
Just want to make sure there's no way we could trigger this error with effectively continuous domains.